concent_api/middleman/middleman_server.py

```diff
@@ -120,52 +124,86 @@ def _close_middleman(self) -> None:
         self._loop.close()
 
     async def _handle_concent_connection(self, reader: asyncio.StreamReader, writer: asyncio.StreamWriter) -> None:
+        tasks = []
+        response_queue = asyncio.Queue(loop=self._loop)
+        self._connection_id = (self._connection_id + 1) % CONNECTION_COUNTER_LIMIT
+        self._response_queue_pool[self._connection_id] = response_queue
         try:
-            frame = await handle_frame_receive_async(reader, settings.CONCENT_PUBLIC_KEY)
-            remote_address = writer.get_extra_info('peername')
-            print("Received %r from %r" % (frame, remote_address))
-            await self._respond_to_user(frame, writer)
-        except (BrokenEscapingInFrameMiddlemanProtocolError, asyncio.LimitOverrunError, MiddlemanProtocolError) as exception:
-            await self._send_immediate_error(writer, exception)
+            request_producer_task = self._loop.create_task(
+                request_producer(self._request_queue, response_queue, reader, self._connection_id)
+            )
+            response_consumer_task = self._loop.create_task(
+                response_consumer(response_queue, writer, self._connection_id)
+            )
+            tasks.append(request_producer_task)
+            tasks.append(response_consumer_task)
+            await request_producer_task
```

Comment:
Could you please add a comment what is the purpose of those three awaits ?

Reply:
Btw. I think there is nothing wrong with putting a comment over the line it is about. Putting it in the same line is ok but might be problematic if:
- the comment is too long (if it is over you can always break it to several lines)
- the comment is about whole block, not just single line
- there will be also some mypy/lint ignore comments.